### PR TITLE
Introduce style guidelines for commit messages

### PR DESCRIPTION
**Pull request by freddieRv**
#### Description:

- Introduce a "Commit messages" section under the "Pull requests" style guidelines

#### Rationale:

- There are quite a few commits on the project with less-than-ideal commit messages
- Having style guidelines enforced to some degree will definitely have a positive impact on the overall maintainability of the project
- Having well-formatted and consistent commit messages makes the log easier to read and use

#### Review Hints:

- Proposing this new guidelines after it was suggested in #10126
- This guidelines are based on [this post](https://cbea.ms/git-commit/#end), which also references a number of relevant places where this practices are recommended

### CODE REVIEW

File: docs/manual/developer/04_style_guide.md [hunk 1]
```diff
@@ -52,6 +52,62 @@ Noteworthy additions to these principles:
 * Should use the [merge commit method](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)
 * Should use the GitHub Web UI to document and ensure code reviews are done correctly
 
+### Commit Messages
+
```
Comment (Mab879) [inline docs/manual/developer/04_style_guide.md:56]: I would mention this is based on the post you linked.

File: docs/manual/developer/04_style_guide.md [hunk 2]
```diff
@@ -52,6 +52,62 @@ Noteworthy additions to these principles:
...
+      to explain what’s going on
+    * If you’re having a hard time summarizing, you might be committing too many changes at once
+* Separate subject from body with a blank line
+    * Not every commit requires both a subject and a body.
+      Sometimes a single line is fine, especially when the change
+      is so simple that no further context is necessary
+* Capitalize the subject line
+* Do not end the subject line with a `.` (period)
+    * Trailing punctuation is unnecessary in subject lines
+* Use the imperative mood
```
Comment (Mab879) [inline docs/manual/developer/04_style_guide.md:70]:
```suggestion
* Use the imperative mood in the subject line
```
I don't think that is needed in the body

**Review by Mab879 — COMMENTED** [review]
Thanks for the PR. Please take look comments and let me know what you think.

File: docs/manual/developer/04_style_guide.md [hunk 3]
```diff
@@ -52,6 +52,62 @@ Noteworthy additions to these principles:
...
+Are there side effects or other unintuitive consequences of this
+change? Here's the place to explain them.
+
+Further paragraphs come after blank lines.
+
+ - Bullet points are okay, too
+ - Typically a hyphen or asterisk is used for the bullet, preceded
+   by a single space
+
+If you use an issue tracker, put references to them at the bottom,
```
[the remark this answers is not in the document]
Comment (freddieRv) [reply docs/manual/developer/04_style_guide.md]: Addressed

File: docs/manual/developer/04_style_guide.md [hunk 4]
```diff
@@ -52,6 +52,62 @@ Noteworthy additions to these principles:
 * Should use the [merge commit method](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges)
 * Should use the GitHub Web UI to document and ensure code reviews are done correctly
 
+### Commit Messages
+
+* Limit the subject line to 50 characters
```
[the remark this answers is not in the document]
Comment (freddieRv) [reply docs/manual/developer/04_style_guide.md]: Addressed

File: docs/manual/developer/04_style_guide.md (hunk 2 above)
Comment (freddieRv) [reply docs/manual/developer/04_style_guide.md]: Addressed

File: docs/manual/developer/04_style_guide.md (hunk 1 above)
Comment (freddieRv) [reply docs/manual/developer/04_style_guide.md:56]: Addressed

**Review by marcusburghardt — APPROVED** [review]
I liked it.

**Pull request merged**